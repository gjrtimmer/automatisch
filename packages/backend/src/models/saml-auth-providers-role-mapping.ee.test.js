import { describe, it, expect } from 'vitest';
import SamlAuthProvidersRoleMapping from '../models/saml-auth-providers-role-mapping.ee';
import SamlAuthProvider from './saml-auth-provider.ee';

describe('SamlAuthProvidersRoleMapping model', () => {
  it('tableName should return correct name', () => {
    expect(SamlAuthProvidersRoleMapping.tableName).toBe(
      'saml_auth_providers_role_mappings'
    );
  });

  it('jsonSchema should have the correct schema', () => {
    expect(SamlAuthProvidersRoleMapping.jsonSchema).toMatchSnapshot();
  });

  it('required properties should be required', async () => {
    expect(SamlAuthProvidersRoleMapping).toRequireProperty('remoteRoleName');
    expect(SamlAuthProvidersRoleMapping).toRequireProperty('roleId');
    expect(SamlAuthProvidersRoleMapping).toRequireProperty(
      'samlAuthProviderId'
    );
  });

  it('relationMappings should have samlAuthProvider relation', () => {
    const samlAuthProviderRelation =
      SamlAuthProvidersRoleMapping.relationMappings().samlAuthProvider;

    expect(samlAuthProviderRelation).toMatchSnapshot();
    expect(samlAuthProviderRelation.relation).toStrictEqual(
      SamlAuthProvidersRoleMapping.BelongsToOneRelation
    );
    expect(samlAuthProviderRelation.modelClass).toStrictEqual(SamlAuthProvider);
  });
});
