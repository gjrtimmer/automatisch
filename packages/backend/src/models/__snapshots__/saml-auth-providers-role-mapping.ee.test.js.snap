// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SamlAuthProvidersRoleMapping model > jsonSchema should have the correct schema 1`] = `
{
  "properties": {
    "id": {
      "format": "uuid",
      "type": "string",
    },
    "remoteRoleName": {
      "minLength": 1,
      "type": "string",
    },
    "roleId": {
      "format": "uuid",
      "type": "string",
    },
    "samlAuthProviderId": {
      "format": "uuid",
      "type": "string",
    },
  },
  "required": [
    "samlAuthProviderId",
    "roleId",
    "remoteRoleName",
  ],
  "type": "object",
}
`;

exports[`SamlAuthProvidersRoleMapping model > relationMappings should have samlAuthProvider relation 1`] = `
{
  "join": {
    "from": "saml_auth_providers_role_mappings.saml_auth_provider_id",
    "to": "saml_auth_providers.id",
  },
  "modelClass": [Function],
  "relation": [Function],
}
`;
